services/src/main/java/io/grpc/services/ChannelzProtoUtil.java

```diff
@@ -348,21 +353,49 @@ static Channel toChannel(Instrumented<ChannelStats> channel) {
   }
 
   static ChannelData extractChannelData(Channelz.ChannelStats stats) {
-    return ChannelData
-        .newBuilder()
-        .setTarget(stats.target)
+    ChannelData.Builder builder = ChannelData.newBuilder();
+    builder.setTarget(stats.target)
         .setState(toChannelConnectivityState(stats.state))
         .setCallsStarted(stats.callsStarted)
         .setCallsSucceeded(stats.callsSucceeded)
         .setCallsFailed(stats.callsFailed)
-        .setLastCallStartedTimestamp(Timestamps.fromMillis(stats.lastCallStartedMillis))
-        .build();
+        .setLastCallStartedTimestamp(Timestamps.fromMillis(stats.lastCallStartedMillis));
+    if (stats.channelTrace != null) {
+      builder.setTrace(toChannelTrace(stats.channelTrace));
+    }
+    return builder.build();
   }
 
   static ChannelConnectivityState toChannelConnectivityState(ConnectivityState s) {
     return ChannelConnectivityState.newBuilder().setState(toState(s)).build();
   }
 
+  private static ChannelTrace toChannelTrace(io.grpc.internal.Channelz.ChannelTrace channelTrace) {
```

Comment:
Can we say `Channelz.ChannelTrace` here?

Reply:
Done.